### PR TITLE
solution att#1.

### PR DESCRIPTION
**Pull request by Manankin**
But not all tests done; continue to fix them

### CODE REVIEW

**Review by YevhenOstrovskiy — CHANGES_REQUESTED** [review]
Looks nice, but not all test have passed!!!
Check how you get id, because you use array length, but if we delete some item and then add another one we can get the same id's. Try to use something like this:
 id: users.length > 0
      ? +(users[users.length - 1].id) + 1 : users.length + 1
      or use 'uuid' library from express instead

File: src/controllers/expense.controller.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+  if (!title || !user) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const newExpense = createExpense({
+    userId,
+    spentAt,
```
Comment (DanilWeda) [inline src/controllers/expense.controller.js:88]: Please fix this lint error or disabled rule

File: src/controllers/expense.controller.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    note,
+  } = req.body;
+
+  if (!title || !category || !note || !amount || !userId || !spentAt) {
+    res.status(400).send('Invalid request body');
+
+    return;
+  }
+
+  const user = getAllUsers().find(u => u.id === +userId);
```
Comment (DanilWeda) [inline src/controllers/expense.controller.js:78]: Fix naming. u - blurry meaning

File: src/controllers/expense.controller.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  } = req.body;
+
+  if (!title || !category || !note || !amount || !userId || !spentAt) {
+    res.status(400).send('Invalid request body');
```
Comment (DanilWeda) [inline src/controllers/expense.controller.js:73]: Add all status codes and message in constants

File: src/services/user.service.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+let users = [];
+let increment = 1;
```
Comment (DanilWeda) [inline src/services/user.service.js:4]: Why do we need let in this case?
Don`t see any reason why we don't use const for this array.
Try rewriting using const. We avoid not needed assignments.

File: src/services/expense.service.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+let expenses = [];
+let increment = 1;
```
Comment (DanilWeda) [inline src/services/expense.service.js:4]: Why do we need let in this case?
Don`t see any reason why we don't use const for this array.
Try rewriting using const. We avoid not needed assignments.

File: src/controllers/user.controller.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  const { name } = req.body;
+  const user = getUserById(+id);
+
+  if (!user) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  if (typeof name !== 'string') {
```
Comment (DanilWeda) [inline src/controllers/user.controller.js:71]: Try not to use type for condition. 
Don`t see any reason for this type of condition. 
Maybe I'm missing something?

File: src/services/expense.service.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+function clearAllExpenses() {
+  expenses = [];
+}
+
+function getAllExpenses() {
+  return expenses;
+}
+
+function getExpenseById(id) {
+  return expenses.find(exp => exp.id === id);
```
Comment (DanilWeda) [inline src/services/expense.service.js:15]: fix naming

File: src/services/expense.service.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+  };
+
+  expenses.push(newExpense);
+  increment++;
+
+  return newExpense;
+}
+
+function deleteExpenseById(id) {
+  const filteredExpenses = expenses.filter(exp => exp.id !== id);
```
Comment (DanilWeda) [inline src/services/expense.service.js:43]: fix naming

**Review by DanilWeda — CHANGES_REQUESTED** [review]
Good job! 
Check some small issues) 
Also, check test suite.
Many thanks!

File: src/controllers/expense.controller.js (hunk 2 above)
Comment (Manankin) [reply src/controllers/expense.controller.js:78]: it's naming inside callback function and I don't use it outside.  guess that it's clear to understand (user => u)

File: src/services/expense.service.js (hunk 5 above)
Comment (Manankin) [reply src/services/expense.service.js:4]: the same reason as above

File: src/services/user.service.js (hunk 4 above)
Comment (DanilWeda) [reply src/services/user.service.js:4]: I see this assignment. 
We can manipulate with an array. 
Any let assigned makes code more dirty readability.
Please try rewrite, also in this case you don't need ` let increment = 1;` part of code.
Thanks!

File: src/services/expense.service.js (hunk 5 above)
Comment (DanilWeda) [reply src/services/expense.service.js:4]: I see this assignment. 
We can manipulate with an array. 
Any let assigned makes code more dirty readability.
Please try rewrite, also in this case you don't need let increment = 1; part of code.
Thanks!

**Review by DanilWeda — CHANGES_REQUESTED** [review]
Check my answer to your comments. 
Thanks)

**Review by DanilWeda — CHANGES_REQUESTED** [review]
Great. All good. 
One small thing for better understanding code. 
Many thanks)

**Review by DanilWeda — APPROVED** [review]
Great! 
Approved. 
Thanks)

**Pull request closed without merging**